include/rocksdb/options.h

```diff
@@ -882,6 +882,16 @@ struct DBOptions {
   // Default: false
   bool enable_pipelined_write = false;
 
+  // If true the write to memtable are done without joining any write thread.
+  // This offers much higher write throughput. The tradeoff is that the writes
+  // visible to a snapshot might change over time. If the application cannot
+  // tolerate that, it should implement its own mechanisms to work around that.
+  // Using TransactionDB with WRITE_PREPARED write policy is one way to achieve
+  // that.
```

Comment:
Can you document the behavior if users don't use transactionDB? What is visible and not visible? It's not very clear to me even after I read the PR.

I feel that it is desirable for users to be able to benefit from the feature even if they don't use transaction DB, but it's not clear to me whether it is possible.

Reply:
This is the best I came up with without making it too complicated. If you have an alternative description paste it here and I will replace.

Reply:
If users don't use transaction DB. They turn unordered_write on, and insert k1=>v1 and k2=>v2. And when they do Get(k1). What result will they get?

By the way, this is a public interface, so "write thread" is an internal concept, which we should try to avoid because they may not understand.

Reply:
I'm almost ready to give the PR an approval, as long as the comment is made clearer.